### ISSUE TITLE
Allow IntervalListTools to subdivide interval list based on number of intervals.

### ISSUE THREAD

**Issue by samuelklee** [broadinstitute/picard#1005]
### Tool(s) involved

IntervalListTools (arguments: SCATTER_COUNT, SUBDIVISION_MODE)

### Description

The current SUBDIVISION_MODE options break up the input interval list into multiple lists based on base length.  A mode that instead breaks up the input list into multiple lists with equal numbers of intervals for scattering CNV analyses would be nice.  This mode should not merge abutting intervals (actually, it is also not clear to me whether the merging of abutting intervals is currently documented correctly).

**Comment by sooheelee** [comment]
@yfarjoun, @jacarey, @cmnbroad, I see you three were actively working on this tool recently, so I've taken the liberty to tag you. 

The CNV workflows (both somatic and germline) could really use this feature. The workflow requires binning intervals into equally sized chunks over WGS data, and we need to maintain these separate adjacent bins and not merge them. To scatter the workflow, we need to then split an intervals list of bins into multiple files while maintaining the separate adjacent bins, again not merging them, which SplitIntervals/IntervalListTools currently does. 

Our current workaround involves the following hack, which we use in [cnv_common_tasks.wdl](https://github.com/broadinstitute/gatk/blob/4.0.4.0/scripts/cnv_wdl/cnv_common_tasks.wdl#L214-L221):
```
command <<<
        set -e

        grep @ ${interval_list} > header.txt
        grep -v @ ${interval_list} > all_intervals.txt
        split -l ${num_intervals_per_scatter} --numeric-suffixes all_intervals.txt ${base_filename}.scattered.
        for i in ${base_filename}.scattered.*; do cat header.txt $i > $i.interval_list; done
    >>>
```

Could we please make this type of functionality a proper tool feature?

**Comment by yfarjoun** [comment]
fixed by #1208
